award_emojis.go

```diff
@@ -0,0 +1,412 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package gitlab
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+)
+
+// AwardEmojiService handles communication with the emoji awards related methods
+// of the GitLab API.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/award_emoji.html
+type AwardEmojiService struct {
+	client *Client
+}
+
+// AwardEmoji represents a GitLab Award Emoji.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/award_emoji.html
+type AwardEmoji struct {
+	ID   int    `json:"id"`
+	Name string `json:"name"`
+	User struct {
+		Name      string `json:"name"`
+		Username  string `json:"username"`
+		ID        int    `json:"id"`
+		State     string `json:"state"`
+		AvatarURL string `json:"avatar_url"`
+		WebURL    string `json:"web_url"`
+	} `json:"user"`
+	CreatedAt     *time.Time `json:"created_at"`
+	UpdatedAt     *time.Time `json:"updated_at"`
+	AwardableID   int        `json:"awardable_id"`
+	AwardableType string     `json:"awardable_type"`
+}
+
+const (
+	awardMergeRequest = "merge_requests"
+	awardIssue        = "issues"
+	awardSnippets     = "snippets"
+)
+
+// ListMergeRequestAwardEmoji gets a list of all award emoji on the merge request
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/award_emoji.html#list-an-awardable-39-s-award-emoji
+func (a *AwardEmojiService) ListMergeRequestAwardEmoji(pid interface{}, mergeRequestIID int, options ...OptionFunc) ([]*AwardEmoji, *Response, error) {
+	return a.listAwardEmoji(pid, awardMergeRequest, mergeRequestIID, options...)
+}
+
+// ListIssueAwardEmoji gets a list of all award emoji on the issue
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/award_emoji.html#list-an-awardable-39-s-award-emoji
+func (a *AwardEmojiService) ListIssueAwardEmoji(pid interface{}, issueIID int, options ...OptionFunc) ([]*AwardEmoji, *Response, error) {
+	return a.listAwardEmoji(pid, awardIssue, issueIID, options...)
+}
+
+// ListSnippetAwardEmoji gets a list of all award emoji on the snippet
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/award_emoji.html#list-an-awardable-39-s-award-emoji
+func (a *AwardEmojiService) ListSnippetAwardEmoji(pid interface{}, snippetID int, options ...OptionFunc) ([]*AwardEmoji, *Response, error) {
+	return a.listAwardEmoji(pid, awardSnippets, snippetID, options...)
+}
+
+func (a *AwardEmojiService) listAwardEmoji(pid interface{}, resource string, resourceID int, options ...OptionFunc) ([]*AwardEmoji, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/%s/%d/award_emoji", url.QueryEscape(project), resource, resourceID)
+
+	req, err := a.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var awardEmoji []*AwardEmoji
```

Comment:
If the method receiver is updated to be `s`, then this could be `as` (or `a` when singular) .This is a convention used in many other locations in this package